src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1540,4 +1604,134 @@ public static void shutdownAndWait(long timeout, TimeUnit units) throws Interrup
         shutdown(asyncExecutor, blockingExecutor);
         awaitTermination(timeout, units, asyncExecutor, blockingExecutor);
     }
+
```

Comment:
> gossip only propagates difference between contacted peers. if index status remains the same, overhead is pretty small.

The way I understand it, is the index status will change when marking it as built (i.e. creating a new index or recovery) or when doing a full rebuild. So yes, I suspect the overhead to be minimal here.

Reply:
yeah, the index status changes are rare